docs/release-process.md

```diff
@@ -2,19 +2,44 @@
 
 The following steps are needed to create a release.
 
-Create a tag on the commit that should be released:
+1. **Generate the Changelog**
+   - Run the following command to generate the changelog on the `main` branch:
 
-```bash
-export VERSION=v0.0.1
-git tag --sign --message="Version ${VERSION}" "${VERSION}"
-```
+     ```bash
+     make changelog
+     ```
 
-Push the tag upstream:
+   - Create a new branch for the release:
 
-```bash
-git push origin "${VERSION}"
-```
+     ```bash
+     export VERSION=v0.0.1
+     git checkout -b <user>/release-${VERSION}
+     ```
```

Comment:
I think the ordering should be 
1. checkout a new release branch `git checkout -b <user>/release-${VERSION}` 
2. bump the `.punch_version.py`
3. `make changelog`
and then merge the PR, switch to `main`, pull changes, and `make release-tag` as described :)

Reply:
> 1. checkout a new release branch git checkout -b <user>/release-${VERSION}

Sure I'll reorder "switching branch" step and running `make changelog`. Although the order of these two steps do not matter.

> 2. bump the .punch_version.py

`make changelog` bumps the `.punch_version.py`. So if you manually bump it before running `make changelog` the version will be increased twice: https://github.com/oasisprotocol/nexus/blob/12a1ea7e900205d33f647b36574a623ec2057009/Makefile#L173